include/dpp/streams.h

```diff
@@ -0,0 +1,158 @@
+/************************************************************************************
+ *
+ * D++, A Lightweight C++ library for Discord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2021 Craig Edwards and D++ contributors 
+ * (https://github.com/brainboxdotcc/DPP/graphs/contributors)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ************************************************************************************/
+#pragma once
+#include <dpp/export.h>
+#include <dpp/user.h>
+#include <dpp/channel.h>
+#include <dpp/cluster.h>
+#include <utility>
+
+
+namespace dpp {
+/**
+ * @brief Return end of message
+ * @return 0
+*/
+int end_msg();
+/**
+ * @brief Simple parent for streams
+ * 
+ */
+
+class DPP_EXPORT base_stream {
+    public:
+        /**
+         * @brief Build message in stream
+         * @param msg Text to add
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const std::string& msg);
+        /**
+         * @brief Send message in stream
+         * @param n Number
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const int& n);
```

Comment:
why only int, why not uint64_t, what about snowflake?

Reply:
The purpose of this is not to put a int into the stream, rather it is to act as an end of message indicator

Reply:
this makes no sense. so I can't send an int with a value of 0 to the stream or it ends the stream?

Reply:
You should pass small variables as a value parameter, not as a const reference.

Reply:
As brain said, this also creates strange behavior, for example :
```cpp
stream << "I have found " << get_count_of_something() << " things" << 0;
```
Will have very different behavior based on whether `get_count_of_something()` returns 0 versus non-0.
I suggest using a tag, such as :
```cpp
class stream {
  struct send_tag {};

  static constexpr send = send_tag{};
  
  base_stream &operator<<(send_tag); // your function that sends
  
  /* rest of the stream class */
};
```

Reply:
why would zero be a special value? the whole point of a stream is it can take lots of types and especially integers

Reply:
(It seems my review comments were added back in time after your replies, so i'm replying here to make sure you see it hehe)

Reply:
Yeah that happened